contracts/settling_game/modules/combat/library.cairo

```diff
@@ -383,131 +304,120 @@ namespace Combat {
     // @param battalions: Number of Battalions
     // @param total_battalions: Total Battalions in Army
     // @param counter_attack: Counter attack value
-    // @param counter_defence: Counter defence value
     // @return new health of battalions and battalions remaining
     //         if health goes to 0, then no battalions are alive and we return 0
-
-    const COMBAT_ALGO_WEIGHT_1 = 50;  // weight in bp
-    const FIXED_DAMAGE_AMOUNT = 20;
-    const BASE_STATISTICS = 7;
-    const BASE_BATTALIONS = 1;  // adds a base value to the battalions to make algo work
-
     func calculate_health_remaining{
         syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
-    }(
-        starting_health: felt,
-        battalions: felt,
-        total_battalions: felt,
-        counter_attack: felt,
-        counter_defence: felt,
-    ) -> (new_health: felt, battalions: felt) {
+    }(starting_health: felt, battalion_attack: felt, counter_defence: felt, hp_loss: felt) -> (
+        new_health: felt, battalions: felt
+    ) {
         alloc_locals;
 
         // get weight of attack over defence
-        let (attack_over_defence, _) = unsigned_div_rem(
-            ((counter_attack + BASE_STATISTICS) * 100) * COMBAT_ALGO_WEIGHT_1,
-            counter_defence + BASE_STATISTICS,
-        );
+        let (attack_over_defence, _) = unsigned_div_rem(battalion_attack * 1000, counter_defence);
 
-        // use weight and multiple by starting health to get remaining health
-        let (health_remaining, _) = unsigned_div_rem(attack_over_defence * starting_health, 10000);
-
-        // get % of calculated battalions over total battalions of that type
-        let (battalion_distribution, _) = unsigned_div_rem(
-            (battalions) * 100, total_battalions + BASE_BATTALIONS
-        );
-
-        // get actual health in of battalion by using the battalion distribution
-        let (real_battalion_health, _) = unsigned_div_rem(
-            health_remaining * battalion_distribution, 100
+        // get base health remaining - divided by 1000000 as values coming in a bp
+        let (health_remaining, _) = unsigned_div_rem(
+            attack_over_defence * hp_loss * starting_health, 1000000
         );
 
-        // add modifier so the health can depleate past 0
-        let modified_health = real_battalion_health - FIXED_DAMAGE_AMOUNT;
-
-        // check if dead,IF yes, then return 0,0
-        let is_dead = is_le(modified_health, 0);
+        // if health 0 the battalion is dead. Return 0,0
+        let is_dead = is_le(actual_health_remaining, 0);
```

Comment:
`actual_health_remaining` is undefined at this point

Reply:
-_-